lib/AdminPages/Pages/Newsletters.php

```diff
@@ -168,6 +168,27 @@ public function render() {
     $data['display_detailed_stats'] = Installer::getPremiumStatus()['premium_plugin_initialized'];
     $data['newsletters_templates_recently_sent_count'] = $this->newsletterTemplatesRepository->getRecentlySentCount();
 
+    $data['product_categories'] = $this->wp->getCategories(['taxonomy' => 'product_cat']);
+
+    usort($data['product_categories'], function ($a, $b) {
+      return strcmp($a->catName, $b->catName);
+    });
+
+    $data['products'] = $this->getProducts();
+
     $this->pageRenderer->displayPage('newsletters.html', $data);
   }
+
+  private function getProducts() {
+    $products = $this->wp->getResultsFromWpDb(
+      "SELECT `ID`, `post_title` FROM {$this->wp->getWPTableName('posts')} WHERE `post_type` = %s ORDER BY `post_title` ASC;",
+      'product'
+    );
+    return array_map(function ($product) {
+      return [
+        'title' => $product->post_title, // phpcs:ignore Squiz.NamingConventions.ValidVariableName.NotCamelCaps
+        'ID' => $product->ID,
+      ];
+    }, $products);
+  }
```

Comment:
We have a bunch of methods in [form editor page controller](https://github.com/mailpoet/mailpoet/blob/master/lib/AdminPages/Pages/FormEditor.php#L358-L415) that do basically the same (maybe without ordering). I think it would be nice to put those into some reusable service and reuse it here and also unify formatted output. With this, we could also avoid additional mapping in views/newsletters.html. I think would be much nicer since now the fetching code is duplicated here and on the forms page and we have 2 slightly different output formats for products and categories. What do you think?